# Homework 1 - NumPy Exercises



In [2]:
import numpy as np


## Task 1: 4×4 Array Operations


In [3]:
np.random.seed(42)
array_4x4 = np.random.randint(1, 51, size=(4, 4))
print("4×4 Array:")
print(array_4x4)


4×4 Array:
[[39 29 15 43]
 [ 8 21 39 19]
 [23 11 11 24]
 [36 40 24  3]]


In [4]:
row_sums = np.sum(array_4x4, axis=1)
column_sums = np.sum(array_4x4, axis=0)
overall_mean = np.mean(array_4x4)

print(f"Row sums: {row_sums}")
print(f"Column sums: {column_sums}")
print(f"Overall mean: {overall_mean:.2f}")


Row sums: [126  87  69 103]
Column sums: [106 101  89  89]
Overall mean: 24.06


## Task 2: Array Transformation


In [5]:
array_3x5 = np.arange(15).reshape(3, 5)
print("Original 3×5 Array:")
print(array_3x5)
print()


Original 3×5 Array:
[[ 0  1  2  3  4]
 [ 5  6  7  8  9]
 [10 11 12 13 14]]



In [6]:
array_transposed = array_3x5.T
print("Transposed Array (5×3):")
print(array_transposed)
print()


Transposed Array (5×3):
[[ 0  5 10]
 [ 1  6 11]
 [ 2  7 12]
 [ 3  8 13]
 [ 4  9 14]]



In [7]:
array_5x3 = array_3x5.reshape(5, 3)
print("Reshaped Array (5×3):")
print(array_5x3)


Reshaped Array (5×3):
[[ 0  1  2]
 [ 3  4  5]
 [ 6  7  8]
 [ 9 10 11]
 [12 13 14]]


## Task 3: Sales Report

In this section, we compute totals for simple data.


In [8]:
sales = [120, 135, 100, 150, 90, 200, 170]

total_sales = sum(sales)
average_sales = sum(sales) / len(sales)
maximum_sales = max(sales)
minimum_sales = min(sales)

print(f"Sales data: {sales}")
print(f"\nTotal sales for the week: {total_sales}")
print(f"Average value: {average_sales:.2f}")
print(f"Maximum value: {maximum_sales}")
print(f"Minimum value: {minimum_sales}")


Sales data: [120, 135, 100, 150, 90, 200, 170]

Total sales for the week: 965
Average value: 137.86
Maximum value: 200
Minimum value: 90


### Results Summary

| Metric | Value |
|--------|-------|
| Total | 965 |
| Average | 137.86 |
| Maximum | 200 |
| Minimum | 90 |
